ESQL: Clarify docs and add csv test for WHERE in STATS (#133629)
elastic/elasticsearch: docs/reference/query-languages/esql/_snippets/commands/layout/stats-by.md

    If its name coincides with one of the computed columns, that column will be ignored.

`boolean_expressionX`
:   The condition that must be met for a row to be included in the evaluation of

>     `expressionX`. Does not have any effect on the values used in

Reviewer: ```suggestion
    `expressionX`. Does not have any effect on the values produced by
```

>     `grouping_expressionX` or the other `expressionX`.

Reviewer: ```suggestion
    `grouping_expressionX` or the other `expressionX`. Consequently, `... | STATS ... WHERE <condition> ...` is not equivalent to `... | WHERE <condition> | STATS ...`.
```

:::{include} ../examples/stats.csv-spec/aggFilteringOnGroup.md
:::

> ### WHERE before STATS excludes non-matching groups

Author: nit: I think the following example is still part of `Filter on the grouping key` (by contrasting the example above). But this is fine, too.
